### PR TITLE
typehints update #3

### PR DESCRIPTION
**Pull request by go-bears**
remove unused mapping in data.py
update transform.py
typehints updates for seq_io files

### CODE REVIEW

File: weblogo/seq_io/clustal_io.py [hunk 1]
```diff
@@ -131,7 +137,7 @@ def read(fin, alphabet=None):
 #     space and then the sequence.
 
 
-def _scan(fin):
+def _scan(fin: TextIO) -> Iterator[Token]:
```
Comment (gecrooks) [inline weblogo/seq_io/clustal_io.py:140]: Return type doesn't look right. Should be Generator?

File: weblogo/seq_io/clustal_io.py [hunk 2]
```diff
@@ -201,7 +207,7 @@ def _scan(fin):
     return
 
 
-def write(fout, seqs):
+def write(fout: TextIO, seqs: List[Seq]) -> None:
```
Comment (gecrooks) [inline weblogo/seq_io/clustal_io.py:210]: Check, List[Seq] or SeqList?

File: weblogo/seq_io/fasta_io.py [hunk 3]
```diff
@@ -93,7 +96,7 @@
 """
 
 
-def read(fin, alphabet=None):
+def read(fin: TextIO, alphabet: Alphabet = None) -> Iterable[Seq]:
```
Comment (gecrooks) [inline weblogo/seq_io/fasta_io.py:99]: Returns SeqList

File: weblogo/seq_io/clustal_io.py (hunk 1 above)
Comment (gecrooks) [reply weblogo/seq_io/clustal_io.py:140]: Actually probably ok. The return statement at 207 makes this look weird. The return isn't needed, remove.

File: weblogo/seq_io/msf_io.py [hunk 4]
```diff
@@ -106,20 +107,23 @@ def read(fin, alphabet=None):
...
+            seqs[block_count].append(data)
             block_count += 1
     if seq_ids == []:
         raise ValueError("Parse error, possible wrong format")
     seqs = [Seq("".join(s), alphabet, name=i) for s, i in zip(seqs, seq_ids)]
     return SeqList(seqs)
 
 
-def _line_is(fin):
+def _line_is(fin: TextIO) -> Generator[Token, None, None]:
```
Comment (gecrooks) [inline weblogo/seq_io/msf_io.py:126]: -> Iterator[Token]

**Review by gecrooks — CHANGES_REQUESTED** [review]

File: weblogo/seq_io/clustal_io.py (hunk 1 above)
Comment (go-bears) [reply weblogo/seq_io/clustal_io.py:140]: remove the return or change return to `-> None` ?

File: weblogo/seq_io/clustal_io.py (hunk 2 above)
Comment (go-bears) [reply weblogo/seq_io/clustal_io.py:210]: I think `SeqList` ? the other write functions in the other modules use `SeqlList` and  `SeqList` makes sense at this point that you're writing out the SeqList after processing?

File: weblogo/seq_io/clustal_io.py (hunk 1 above)
Comment (gecrooks) [reply weblogo/seq_io/clustal_io.py]: Remove the return statement on line 207. Not needed I think.

File: weblogo/seq_io/clustal_io.py (hunk 2 above)
Comment (gecrooks) [reply weblogo/seq_io/clustal_io.py]: SeqList to be consistent then.

File: weblogo/seq_io/clustal_io.py (hunk 1 above)
Comment (go-bears) [reply weblogo/seq_io/clustal_io.py]: ooh! got it. Thanks!

**Review by gecrooks — APPROVED** [review]

**Comment by go-bears** [comment]
👍  🚀

**Pull request merged**